tests/testcases/030_check-network.yml

```diff
@@ -15,6 +15,30 @@
       bin_dir: "/usr/local/bin"
     when: not ansible_os_family in ["CoreOS", "Coreos", "Container Linux by CoreOS", "Flatcar", "Flatcar Container Linux by Kinvolk"]
 
+  - name: Approve kubelet serving certificates
+    block:
+
+    - name: Get certificate signing requests
+      command: "{{ bin_dir }}/kubectl get csr -o name"
+      register: get_csr
+      changed_when: false
+
+    - name: Check there are csrs
+      assert:
+        that: get_csr.stdout_lines | length > 0
+        fail_msg: kubelet_rotate_server_certificates is {{ kubelet_rotate_server_certificates }} but no csr's found
+
+    - name: Approve certificates
+      command: "{{ bin_dir }}/kubectl certificate approve {{ get_csr.stdout_lines | join(' ') }}"
```

Comment:
I don't really like how this is implemented. What I was expecting in this PR was the following:
1 - needed vars
2 - logic to enable cert renewal in kubelet config
3 - renewal logic

What I see here is kubectl get csr/kubectl certificate approve in the tests directory, not part of the actual Kubespray core Ansible playbook. Why can't we put this in kubernetes/kubeadm role directly and run it on the first controller? 

From a security perspective, we should filter csrs for those that belong to nodes we know about from inventory, rather than blindly approving all.

Reply:
Thanks for the review.

ad 1 - there's var `kubelet_rotate_server_certificates` that is documented
ad 2 - that's kubelet config parameter `ServerTLSBootstrap`
ad 3 - both bootstrap and renewal are handled by kubelet when the parameter in 2 is set

I would recommend not putting approval logic in kubespray. There is [a reason](https://github.com/kubernetes/community/pull/1982) why serving certificates aren't automatically approved by CSR approving controller like client certificates are. Cluster admin is supposed to manually approve them (or implement a custom approving controller). In any case, approvals are not in scope of this PR. That is why the functionality is disabled by default and the docs warn about the caveat.

Regarding blind approvals in the test, that part takes the role of the cluster admin manually approving certificates and was added just to pass the test which failed otherwise (`kubectl exec` couldn't connect to the kubelet which was missing the serving certificate).